plugin/evm/vm.go

```diff
@@ -457,12 +448,12 @@ func (vm *VM) ParseBlock(b []byte) (snowman.Block, error) {
 	}
 	blockHash := ethBlock.Hash()
 	// Coinbase must be zero on C-Chain
-	if bytes.Compare(blockHash.Bytes(), vm.genesisHash.Bytes()) != 0 &&
-		bytes.Compare(ethBlock.Coinbase().Bytes(), coreth.BlackholeAddr.Bytes()) != 0 {
+	if !bytes.Equal(blockHash.Bytes(), vm.genesisHash.Bytes()) &&
+		!bytes.Equal(ethBlock.Coinbase().Bytes(), coreth.BlackholeAddr.Bytes()) {
```

Comment:
Can we just use `!=` here?

Reply:
No, byte slices can only be compared to nil. It looks like the comparison between an ID or byte array is valid with `==` or `!=` operators, but byte slices require `bytes.Equal(a,b)`.

Reply:
```
if blockHash != vm.genesisHash && ethBlock.Coinbase() != coreth.BlackholeAddr {
```

Reply:
Ah ok, just pushed a commit to udpate this.